.github/workflows/ci.yml

```diff
@@ -0,0 +1,167 @@
+
+      - name: Check for differences
+        run: git --no-pager diff --color --exit-code
+
+  codegen:
+    name: Check CubeMX Code Gen
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        board:
+          - FSM
+          - PDM
+          - BMS
+          - DCM
+          - DIM
+          - GSM
+
+    container:
+      image: ubcformulaelectric/ci:latest
+      credentials:
+        username: ubcformulaelectric
+        password: ${{ secrets.DOCKER_TOKEN }}
+
+    steps:
+      # Fix to get git diff working in a Docker container
+      # https://github.com/actions/checkout/issues/363
+      - name: Setup git
+        run: git config --global --add safe.directory $(realpath .)
+
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      # xvfb-run is required to prefix any commands that run STM32CubeMX
+      # It fakes out a display, since STM32CubeMX throws an error if it doesn't detect a display
+      - name: Generate code
+        run: |
+          xvfb-run python3 scripts/utilities/generate_cube_code.py \
+            --board ${{ matrix.board }} \
+            --log4j_properties_output ~/.stm32cubemx/log4j.properties \
+            --ioc boards/${{ matrix.board }}/${{ matrix.board }}.ioc \
+            --codegen_output_dir boards/${{ matrix.board }} \
+            --cube_bin /usr/local/STM32CubeMX/STM32CubeMX
+          
+      - name: Fix formatting
+        run: python3 clang-format/fix_formatting.py
+
+      - name: Check for differences
+        run: git --no-pager diff --color --exit-code
+
+  build-firmware:
+    name: Build ARM Binaries
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        board:
+          - FSM
+          - PDM
+          - BMS
+          - DCM
+          - DIM
+          - GSM
+
+    container:
+      image: ubcformulaelectric/ci:latest
+      credentials:
+        username: ubcformulaelectric
+        password: ${{ secrets.DOCKER_TOKEN }}
+
+    env:
+      BUILD_DIR: boards/build_arm
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      # xvfb-run is required to prefix any commands that run STM32CubeMX
+      # It fakes out a display, since STM32CubeMX throws an error if it doesn't detect a display
+      # Don't use Pipenv for Cmake since python packages are installed system-wide in the container
+      - name: Build
+        run: |
+          cmake -S boards -B $BUILD_DIR -DPLATFORM=arm -DNO_VENV=ON
+          xvfb-run make --directory=$BUILD_DIR -j`nproc` ${{ matrix.board }}.elf
+
+      # Uploads .elf binary to Github Actions workflow
+      - name: Save built artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.board }}.elf
+          path: ${{ env.BUILD_DIR }}/${{ matrix.board }}/${{ matrix.board }}.elf
+  
+  test:
+    name: Build and Run Tests
```

Comment:
I like that we're separating out the x86 tests per board (easier to debug) but I think you missed the shared tests?

Reply:
yup, added them in now
also added memcheck in for the tests which I forgot about before